test/test-Storage.js

```diff
@@ -0,0 +1,89 @@
+"use strict";
+
+const {before, after} = require("sdk/test/utils");
+const {Storage} = require("lib/task-queue/Storage.js");
+
+let STORAGE;
+
+exports["test that tables are created"] = function*(assert) {
+  let tableCheckSQL = "SELECT name FROM sqlite_master WHERE type='table' AND name=:name";
+  let pagesTable = yield STORAGE.executeCached("check pages table existence", tableCheckSQL, {
+    params: {name: "moz_pages"}
+  });
+  assert.ok(pagesTable.length, "Found a moz_pages table");
+
+  let tasksTable = yield STORAGE.executeCached("check tasks table existence", tableCheckSQL, {
+    params: {name: "moz_tasks"}
+  });
+  assert.ok(tasksTable.length, "Found a moz_tasks table");
+};
+
+exports["test inserting, selecting and deleting a page"] = function*(assert) {
+  const insertPage = `INSERT INTO moz_pages (url, path, expiry, createdAt, remote)
+                      VALUES (:url, :path, :expiry, :createdAt, :remote);`;
+  let params = {
+    url: "http://foo.bar",
+    path: "1234",
+    expiry: 1467225044,
+    createdAt: null,
+    remote: 0};
+
+  let noPage = yield requestPage({url: params.url});
+  assert.equal(noPage.length, 0, "The table is empty.");
+
+  yield STORAGE.executeCached("add entry to pages", insertPage, {params});
+  let page = yield requestPage({url: params.url});
+
+  assert.equal(page.length, 1, "The table has 1 entry.");
+  assert.deepEqual(page, [params], "Selected page is equal to the inserted one.");
+
+  function*requestPage(params) {
```

Comment:
I think you need to wrap that function  in `Task.spawn`, then you can yield it like so:

```javascript
let page = yield Task.spawn(function*(url) {
...
});
```